tests/functional/Codeception/Docker.php

```diff
@@ -0,0 +1,261 @@
+     */
+    public function _initialize()
+    {
+        $container = require Configuration::projectDir() . 'tests/functional/bootstrap.php';
+        $builder = CollectionBuilder::create($container, $this);
+
+        $this->setContainer($container);
+        $this->setBuilder($builder);
+    }
+
+    /**
+     * @inheritdoc
+     */
+    public function _before(TestInterface $test)
+    {
+        $this->taskEnvUp()
+            ->run()
+            ->stopOnFail();
+    }
+
+    /**
+     * @inheritdoc
+     */
+    public function _after(TestInterface $test)
+    {
+        $this->taskEnvDown()
+            ->run()
+            ->stopOnFail();
+    }
+
+    /**
+     * Clones magento cloud template from git
+     *
+     * @param string|null $version
+     * @return bool
+     * @throws \Robo\Exception\TaskException
+     */
+    public function cloneTemplate(string $version = null): bool
+    {
+        $gitTask = $this->taskGitStack()
+            ->exec('git init')
+            ->exec('git remote -vvv')
+            ->exec(sprintf('git remote add origin %s', $this->_getConfig('repo_url')))
+            ->exec('git fetch')
+            ->checkout($version ?: $this->_getConfig('repo_branch'));
+
+        return $this->taskBash(self::BUILD_CONTAINER)
+            ->exec($gitTask)
+            ->run()
+            ->wasSuccessful();
+    }
+
+    /**
+     * Runs composer require command
+     *
+     * @param string $version
+     * @return bool
+     * @throws \Robo\Exception\TaskException
+     */
+    public function composerRequireMagentoCloud(string $version): bool
+    {
+        $composerRequireTask = $this->taskComposerRequire('composer')
+            ->dependency('magento/magento-cloud-metapackage', $version)
+            ->workingDir($this->_getConfig('system_magento_dir'))
+            ->noInteraction()
+            ->option('--no-update');
+        $composerUpdateTask = $this->taskComposerUpdate('composer');
+
+        return $this->taskBash(self::BUILD_CONTAINER)
+            ->exec($composerRequireTask->getCommand() . ' && ' . $composerUpdateTask->getCommand())
+            ->run()
+            ->wasSuccessful();
+    }
+
+    /**
+     * Runs composer install command
+     *
+     * @return bool
+     * @throws \Robo\Exception\TaskException
+     */
+    public function composerInstall(): bool
+    {
+        $composerTask = $this->taskComposerInstall('composer')
+            ->noDev()
+            ->noInteraction()
+            ->workingDir($this->_getConfig('system_magento_dir'));
+        return $this->taskBash(self::BUILD_CONTAINER)
+            ->exec($composerTask)
+            ->run()
+            ->wasSuccessful();
+    }
+
+    /**
+     * Cleans directories
+     *
+     * @param string|array $path
+     * @param string $container
+     * @return bool
+     * @throws \Robo\Exception\TaskException
+     */
+    public function cleanDirectories($path, string $container = self::BUILD_CONTAINER): bool
```

Comment:
According to the method name, it should accept some array of paths not a single path.
It's not a good approach to build logic on the provided argument type.

Looks like it's better to create two methods `cleanDirectory` and `cleanDirectories` or accept only arrays as an argument.

Reply:
Good point.
I'll think about it.